src/_pytest/capture.py

```diff
@@ -579,7 +526,7 @@ def _start(self):
 
     def snap(self):
         self.tmpfile.seek(0)
-        res = self.tmpfile.read()
+        res = self.tmpfile.buffer.read()
```

Comment:
This I think is a bug fix anyway?

Reply:
Maybe fixes https://github.com/pytest-dev/pytest/issues/6871 ?
Might make sense to pull it out then already.

Reply:
But would also avoid the indirection via `__getattr__`, no?